pico/processor/src/main/java/io/helidon/pico/processor/ActiveProcessorUtils.java

```diff
@@ -0,0 +1,308 @@
+/*
+ * Copyright (c) 2023 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.pico.processor;
+
+import java.io.File;
+import java.net.URI;
+import java.nio.file.Path;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.ServiceLoader;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.Predicate;
+
+import javax.annotation.processing.AbstractProcessor;
+import javax.annotation.processing.Filer;
+import javax.annotation.processing.ProcessingEnvironment;
+import javax.annotation.processing.RoundEnvironment;
+import javax.lang.model.element.TypeElement;
+import javax.lang.model.type.TypeMirror;
+import javax.tools.Diagnostic;
+import javax.tools.FileObject;
+import javax.tools.StandardLocation;
+
+import io.helidon.builder.processor.spi.TypeInfoCreatorProvider;
+import io.helidon.builder.processor.tools.BuilderTypeTools;
+import io.helidon.common.HelidonServiceLoader;
+import io.helidon.common.types.TypeInfo;
+import io.helidon.common.types.TypedElementName;
+import io.helidon.pico.tools.Messager;
+import io.helidon.pico.tools.ModuleInfoDescriptor;
+import io.helidon.pico.tools.Options;
+import io.helidon.pico.tools.ServicesToProcess;
+
+import static io.helidon.pico.processor.GeneralProcessorUtils.toPath;
+import static io.helidon.pico.tools.ModuleUtils.REAL_MODULE_INFO_JAVA_NAME;
+import static io.helidon.pico.tools.ModuleUtils.inferSourceOrTest;
+
+/**
+ * Carries methods that are relative only during active APT processing.
+ *
+ * @see GeneralProcessorUtils
+ */
+final class ActiveProcessorUtils implements Messager {
+    static final boolean MAYBE_ANNOTATIONS_CLAIMED_BY_THIS_PROCESSOR = false;
+    static final String TARGET_DIR = "/target/";
+    static final String SRC_MAIN_JAVA_DIR = "/src/main/java";
+
+    private final System.Logger logger;
+    private final ProcessingEnvironment processingEnv;
+    private final TypeInfoCreatorProvider typeInfoCreatorProvider;
+    private RoundEnvironment roundEnv;
+
+    ActiveProcessorUtils(AbstractProcessor processor,
+                         ProcessingEnvironment processingEnv,
+                         RoundEnvironment roundEnv) {
```

Comment:
It seems that the `roundEnv` is set to `null` when constructing, and then updated using a setter method. 
Please remove it from constructor

Reply:
good catch - addressed.